src/v/cluster/feature_table.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/types.h"
+
+#include <string_view>
+
+class cluster_version;
+
+namespace cluster {
+
+enum class feature {
+    central_config = 0x1,
+};
+
+std::string_view to_string_view(feature);
+
+using feature_list = std::vector<feature>;
+
+/**
+ * To enable all shards to efficiently check enablement of features
+ * in their hot paths, the cluster logical version and features
+ * are copied onto each shard.
+ *
+ * Instances of this class are updated by feature_manager.
+ */
+class feature_table {
+public:
+    cluster_version get_active_version() const noexcept {
+        return _active_version;
+    }
+
+    feature_list get_active_features() const;
```

Comment:
since the internal implementation uses bitset to store the features maybe we could just return the bitset here ?

Reply:
The idea is to keep the bitset hidden as an implementation detail of the class, because it might later change to have a more structured per-storage feature as/when we extend feature management to include manual toggling of features (is it enabled, was it ever enabled).  In this particular case, the function is only used for the admin API, which would end up converting the bitset to a list immediately anyway.